### DISCUSSION TITLE
how to use client side code along with server code

### DISCUSSION THREAD

**Discussion by AbyerAli** [vercel/next.js#55156] [App Router]
i am bit new to next js environment and i've started out with the new app router.
now is it possible to render intial data at server and then have the same dynamic touch of rect on client side ? how's it possible in  app router ? or is it even supported yet ?

**Comment by icyJoseph** [comment]
> and then have the same dynamic touch of rect on client side 

What does this mean? Did you mean: `and then have the same dynamic touch of react on client side`

If that's the case, then yah for sure, that's what the so called client components are for. 

> https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs

You can have a Server Component parent that fetches data server side, and passes it to a Client component that injects interactivity. 

Here you can see an example: 

- https://github.com/icyJoseph/next-js-react-gbg/blob/feat/app-router-rsc/demo-app/src/app/collection/page.tsx 

That component page gets some server side data and passes it to:

- https://github.com/icyJoseph/next-js-react-gbg/blob/feat/app-router-rsc/demo-app/src/app/collection/collection.tsx

Which does the interactive stuff.
